### ISSUE TITLE
Consider forcing "pickle" serialization protcol for Dask executor

### ISSUE THREAD

**Issue by cjao** [AgnostiqHQ/covalent#648]
Dask normally uses a [custom serialization protocol](https://distributed.dask.org/en/stable/serialization.html#defaults) for passing data between processes. This protocol fails to preserve some Python data types, leading to bugs such as #638. As a short term fix we can force Dask to use the "pickle" protocol pending corrections to the "dask" protocol. We will of course need to verify that other workflows continue to run with this change.

**Comment by cjao** [comment]
A major strike against "pickle" is that it fails to preserve PyTorch gradients:

```
import torch
from dask.distributed.protocol import serialize, deserialize

def loss(x):
    l = torch.sum(x**2)
    return l

x = torch.tensor([2.0], requires_grad=True)
cost = loss(x)

cost.backward()

print(x, type(x), x.grad)
```
```
>>> tensor([2.], requires_grad=True) <class 'torch.Tensor'> tensor([4.])
```

```
y = deserialize(*serialize(x, serializers=["pickle"]))

print(y, type(y), y.grad)
```
```
>>> tensor([2.], requires_grad=True) <class 'torch.Tensor'> None
```
But the default "dask" protocol actually works here:
```
y = deserialize(*serialize(x, serializers=["dask"]))

print(y, type(y), y.grad)
```
```
>>> tensor([2.], requires_grad=True) <class 'torch.Tensor'> tensor([4.])
```

**Comment by cjao** [comment]
Obsolete.
